reviewers/ESLINTChecker.js

```diff
@@ -0,0 +1,265 @@
+      'consistent-return': this.SEVERITY_HIGH,
+      'curly': this.SEVERITY_HIGH,
+      'dot-notation': this.SEVERITY_HIGH,
+      'eqeqeq': this.SEVERITY_HIGH,
+      'no-eval': this.SEVERITY_MEDIUM,
+      'no-lone-blocks': this.SEVERITY_HIGH,
+      'no-loop-func': this.SEVERITY_HIGH,
+      'no-magic-numbers': this.SEVERITY_HIGH,
+      'no-multi-spaces': this.SEVERITY_HIGH,
+      'no-native-reassign': this.SEVERITY_HIGH,
+      'no-new': this.SEVERITY_HIGH,
+      'no-new-func': this.SEVERITY_HIGH,
+      'no-proto': this.SEVERITY_HIGH,
+      'no-redeclare': this.SEVERITY_HIGH,
+      'no-return-assign': this.SEVERITY_HIGH,
+      'no-self-assign': this.SEVERITY_HIGH,
+      'no-self-compare': this.SEVERITY_HIGH,
+      'no-unmodified-loop-condition': this.SEVERITY_HIGH,
+      'no-unused-expressions': this.SEVERITY_HIGH,
+      'no-useless-call': this.SEVERITY_HIGH,
+      'no-useless-concat': this.SEVERITY_HIGH,
+      'no-useless-escape': this.SEVERITY_HIGH,
+      'no-shadow': this.SEVERITY_LOW,
+      'no-shadow-restricted-names': this.SEVERITY_LOW,
+      'array-bracket-spacing': [this.SEVERITY_LOW, 'never'],
+      'block-spacing': this.SEVERITY_LOW,
+      'brace-style': this.SEVERITY_LOW,
+      'camelcase': this.SEVERITY_LOW,
+      'comma-spacing': this.SEVERITY_LOW,
+      'consistent-this': [ this.SEVERITY_LOW, 'me'],
+      'func-names': this.SEVERITY_LOW,
+      'indent': [this.SEVERITY_LOW, 2],
+      'key-spacing': this.SEVERITY_LOW,
+      'keyword-spacing': this.SEVERITY_LOW,
+      'quotes': [this.SEVERITY_LOW, 'single'],
+      'semi': this.SEVERITY_LOW,
+      'semi-spacing': this.SEVERITY_LOW
+    }
+
+    // Lint the base source.
+    errors.base = linter.verify( baseSource, {
+      rules: rules 
+    });
+    
+    // Lint the head source.
+    errors.head = linter.verify( headSource, {
+      rules: rules 
+    });
+    
+    // Remember this report.
+    var report = {
+      file: to,
+      errors: errors
+    };
+    
+    this.checkedFiles.push( report );
+  },
+  
+  /**
+   * Processes a step in the diff file.
+   * 
+   * @param change      Line being read
+   * @param path        File path
+   * @param position    Position in file
+   * @param callback    Once processing is done.
+   */
+  step: function ESLintStep( change, path, position, callback ) {
+    var comment = '';
+    
+    // On the line, only report on changes that were made.
+    if ( change.add ) {
+      for ( j = 0; j < this.checkedFiles.length; j++ ) {
+        var f = this.checkedFiles[j]; 
+        if ( f.file === path ) {
+          for ( i = 0; i < f.errors.head.length; i++ ) {
+            if ( f.errors.head[i] !== null && f.errors.head[i].line === change.ln ) {
+              comment += f.errors.head[i].message + '\n```javascript\n' + f.errors.head[i].source + '\n```\n';
+              f.errors.head[i].reported = true;
+            }
+          }
+        }
+      }
+    }
+    
+    callback( comment );
+  },
+  
+  /**
```

Comment:
Missing JSDoc parameter type for 'callback'.
```javascript
  /**
```
Missing JSDoc @returns for function.
```javascript
  /**
```